configure.ac

```diff
@@ -1696,6 +1700,26 @@ AC_SUBST(LAPACKINC)
 
 # Done checking LAPACK
 
+# libsvm option
+LIBSVM_INC=
+LIBSVM_LIB=
+USE_LIBSVM=
+
+LOC_CHECK_USE(libsvm,libsvm,USE_LIBSVM)
+
+if test -n "$USE_LIBSVM"; then
+  LOC_CHECK_INC_PATH(libsvm,libsvm,LIBSVM_INC)
+  LOC_CHECK_INCLUDES(libsvm/svm.h,libsvm,$LIBSVM_INC)
```

Comment:
> I replaced with a better check – it first checks for "libsvm/svm.h" and then only for plain "svm.h" before giving up.

Even better! 👍

Reply:
But I suspect you need to change the includes to `#include <svm.h>` anyway, but pass a new "SVM_INCLUDES" variable to the make files. Which may look like `-I$prefix/include/libsvm`.

Reply:
Using $LIBSVM_INC should do I believe.

Reply:
Sorry! I made notes here before pushing changes. I'll run tests with moving svm.h file around to see if compilation succeeds in various configurations.

Reply:
If the include looks like `#include <svm.h>`, that means you need to configure it to point to `/usr/include/libsvm` every time even when it is in a fairly standard location, no? Or is the nesting to libsvm figured out automatically?

Reply:
The path ($LIBSVM_INC) needs to be passed via Platform.make.in to the makefiles that needs them.

Reply:
Apparently there is no easy way how to determine in which directory header file was found in the default search path. I implemented a workaround but I am not too happy about it. Is there any known system where svm.h is not installed into a libsvm directory (installs outside of default search path excluded)?

Reply:
> Apparently there is no easy way how to determine in which directory header file was found in the default search path. I implemented a workaround but I am not too happy about it.

If you could wait a couple of days, I’ll give it a try if you don’t mind.

> Is there any known system where svm.h is not installed into a libsvm directory (installs outside of default search path excluded)?

There is, MacPorts is at least one.

Reply:
> > Is there any known system where svm.h is not installed into a libsvm directory (installs outside of default search path excluded)?
> 
> There is, MacPorts is at least one.

And apparently FreeBSD too (https://cgit.freebsd.org/ports/tree/science/libsvm/pkg-plist).

Reply:
If it's okey by you, I can push a commit with a slightly less complicated configure solution.

Reply:
> If it's okey by you, I can push a commit with a slightly less complicated configure solution.

Yes, if you managed to make a better solution, feel free to push it in.

Reply:
Just pushed my suggestion, please pull and test.

Reply:
Your approach doesn't seem to work as in the case of `svm.h` being located in `/usr/include/libsvm/` directory, it adds a non-existing directory `-I/usr/local/include/libsvm` instead of the directory with svm.h file.

Reply:
In that case I suspect the use of `${includedir}` variable in line 1720, is not the proper one...

Reply:
Could you take a look at `config.log` if there is a more likely candidate for using as variable?

Reply:
The `includedir` is GRASS include directory as it is defined as `includedir='${prefix}/include'` (also as stated in the autoconf documentation).
My approach also wasn't the best but I was on a correct path. I changed my version to use $CC in a way that works with GCC and CLANG compilers (probably others too).

Reply:
Yes, this also works. So many ways how to solve the problem. Got a bit of hair pulling till understood the importance of `config.h.in` file ;-)